src/cpp/server/secure_server_credentials.h

```diff
@@ -61,21 +55,12 @@ class AuthMetadataProcessorAsyncWrapper final {
 
 class SecureServerCredentials final : public ServerCredentials {
```

Comment:
Please add a TODO here to remove this class once we either (a) allow `AuthMetadataProcessor` to be used with any creds type as requested in #21589 or (b) find a way to remove `AuthMetadataProcessor` in favor of some new server-side interception API.